src/components/page-layout/components/exampleIndex.js

```diff
@@ -0,0 +1,407 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import CardContainer from '../../card-container/card-container';
+import { AsideHeading } from '../../on-this-page/helpers';
+import Card from '../../card/card';
+import classnames from 'classnames';
+import ControlSwitch from '@mapbox/mr-ui/control-switch';
+import ControlSelect from '@mapbox/mr-ui/control-select';
+import { ContentWrapper } from './content';
+
+// options of frontMatter.showFilters
+export const filterOptions = [
+  'products',
+  'topics',
+  'languages',
+  'levels',
+  'videos'
+];
+
+export default class ExampleIndex extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      topic: undefined,
+      language: undefined,
+      level: undefined,
+      videos: false,
+      product: undefined
+    };
+  }
+
+  handleInput = (value, id) => {
+    const obj = {};
+    obj[id] = value;
+    this.setState(obj, () => {
+      this.setQueryString(id, value);
+    });
+  };
+
+  componentDidMount() {
+    const { filters } = this.props;
+    // if url query param is set then update the state
+    if ('URLSearchParams' in window) {
+      const query = new URLSearchParams(window.location.search);
+      if (query.get('topic')) {
+        this.setStateIfValid(query.get('topic'), filters.topics, 'topic');
+      }
+      if (query.get('product')) {
+        this.setStateIfValid(query.get('product'), filters.products, 'product');
+      }
+      if (query.get('language')) {
+        this.setStateIfValid(
+          query.get('language'),
+          filters.languages,
+          'language'
+        );
+      }
+      if (query.get('level')) {
+        this.setStateIfValid(
+          parseInt(query.get('level')),
+          filters.levels,
+          'level'
+        );
+      }
+      if (query.get('videos')) {
+        this.setStateIfValid(query.get('videos') === 'true', [true], 'videos');
+      }
+    }
+  }
+
+  // make sure query value is valid before setting the state
+  setStateIfValid = (value, options, id) => {
+    // if query string value is valid, set the state
+    if (options.indexOf(value) > -1) {
+      const obj = {};
+      obj[id] = value;
+      this.setState(obj);
+    } else {
+      // otherwise remove the query string
+      this.setQueryString(id, undefined);
+    }
+  };
+
+  // set the query string or delete it if value is undefined
+  setQueryString = (label, value) => {
+    if ('URLSearchParams' in window) {
+      let searchParams = new URLSearchParams(window.location.search);
+      if (value) searchParams.set(label, value);
+      else searchParams.delete(label);
+      history.pushState(
+        null,
+        '',
+        window.location.pathname + '?' + searchParams.toString()
+      );
+    }
+  };
```

Comment:
I'm not sure we've used the URLSearchParams API in docs before! Here are some links if it's helpful:

- https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams (This API says there is no IE support, but I tested in IE11 and it works 🤷. But since on line 86, we check that `URLSearchParams` exists in window, if a browser didn't support the API, it would skip the logic and not push the URL. We can consider this feature a progressive enhancement.)
- https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

Reply:
I added some more inline comments to help describe this function.